### PR TITLE
Feat/modeling notebook

### PR DESCRIPTION
**Pull request by slebastard**
This branch adds notebook `01_e2e_training.ipynb` to allow fast and flexible testing of any model, starting from our default model configuration: variable list, pre-processing pipelines, model...

In the future, this may:
- replace `00-getting_started.ipnb`, depending on what we want to put in each notebook
- end up on a private repository if we judge that the notebook is most often used for trying out models for which sharing detailed results, including private informations, would be useful

@vviers WDYT?

### CODE REVIEW

File: predictsignauxfaibles/utils.py [hunk 1]
```diff
@@ -1,6 +1,9 @@
 import logging
 from typing import NamedTuple, List
 
+import importlib.util
+from pathlib import Path
```
Comment (vviers) [inline predictsignauxfaibles/utils.py]: Minor detail but please look at the [PEP8 rule for import order](https://www.python.org/dev/peps/pep-0008/#:~:text=Imports%20are%20always%20put%20at,Standard%20library%20imports.) :

> Imports should be grouped in the following order:
> - Standard library imports.
> - Related third party imports.
> - Local application/library specific imports.
>
> You should put a blank line between each group of imports.

File: predictsignauxfaibles/utils.py [hunk 2]
```diff
@@ -173,3 +176,21 @@ def set_if_not_none(obj, attr, val):
...
+    conf_filepath = Path("../models") / model_name / "model_conf.py"
+    if not conf_filepath.exists():
+        raise ValueError(f"{conf_filepath} does not exist")
+
+    spec = importlib.util.spec_from_file_location(
+        f"models.{model_name}.model_conf", conf_filepath
+    )
+    model_conf = importlib.util.module_from_spec(spec)
+    spec.loader.exec_module(model_conf)  # pylint: disable=wrong-import-position
+    return model_conf
```
Comment (vviers) [inline predictsignauxfaibles/utils.py:194]: perfect 👍🏼 

I think this function already existed in `__main__.py` right ? If so, let's remove it from there to avoir redundancy :)

File: notebooks/01-e2e_training.ipynb [hunk 3]
```diff
@@ -0,0 +1,382 @@
...
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "train.fetch_data().raise_if_empty()\n",
+    "test.fetch_data().raise_if_empty()\n",
+    "logging.info(\"Succesfully loaded Features data from MongoDB\")\n",
+    "\n",
+    "if True:\n",
```
Comment (vviers) [inline notebooks/01-e2e_training.ipynb]: `if True:` is pretty useless 😉

File: notebooks/01-e2e_training.ipynb [hunk 4]
```diff
@@ -0,0 +1,382 @@
...
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "train_filepath = \"/path/to/train_dataset.csv\"\n",
+    "test_filepath = \"/path/to/test_dataset.csv\"\n",
+    "\n",
+    "train.data = pd.read_csv(train_filepath)\n",
+    "test.data = pd.read_csv(test_filepath)\n",
+    "\n",
+    "logging.info(\"Succesfully loaded Features data from %s\", features_filepath)"
```
Comment (vviers) [inline notebooks/01-e2e_training.ipynb]: ideally use f-string interpolation here

**Review by vviers — DISMISSED** [review]
LGTM

File: predictsignauxfaibles/utils.py (hunk 1 above)
Comment (slebastard) [reply predictsignauxfaibles/utils.py]: Got it. If I'm not mistaken, everything here is standard library, so I'll sort everything in one unique group

**Review by slebastard — COMMENTED** [review]
Added minor changes based on @vviers's review, merging.

**Pull request merged**